src/coreclr/inc/profilepriv.inl

```diff
@@ -1762,6 +1762,23 @@ inline void ProfControlBlock::EventPipeProviderCreated(EventPipeProvider *provid
                        provider);
 }
 
+inline HRESULT EnumerateGCHeapObjectsCallbackHelper(EEToProfInterfaceImpl *profInterface, ObjectCallback callback, void* callbackState)
+{
+    return profInterface->EnumerateGCHeapObjectsCallback(callback, callbackState);
+}
+
+inline HRESULT ProfControlBlock::EnumerateGCHeapObjectsCallback(ObjectCallback callback, void* callbackState)
+{
+    LIMITED_METHOD_CONTRACT;
+
+    return DoProfilerCallback(ProfilerCallbackType::Active,
```

Comment:
This will iterate all attached profilers and invoke the callback for each one of them. I do not think it is what we want for the `EnumerateGCHeapObjects` API. The callback should be only called for the one profiler that called the API.